synth/src/sampler.rs

```diff
@@ -72,42 +93,44 @@ struct NamespaceSampleStrategy {
 }
 
 impl NamespaceSampleStrategy {
-    fn sample<R: Rng>(&self, mut model: Aggregate<Graph>, mut rng: R) -> Result<Value> {
+    fn sample<R: Rng>(self, model: Graph, mut rng: R) -> Result<Vec<(String, Vec<Value>)>> {
         let mut generated = 0;
-        let mut out = BTreeMap::new();
+        let mut out = BTreeMap::<String, Vec<Value>>::new();
         let progress_bar = sampler_progress_bar(self.target as u64);
 
+        let ordered: Vec<_> = model
+            .iter_ordered()
+            .map(|iter| iter.map(|s| s.to_string()).collect())
+            .unwrap_or_else(Vec::new);
+
+        let mut model = model.aggregate();
+
         while generated < self.target {
             let next = model.complete(&mut rng)?;
-            // Here we are building a BTreeMap.
```

Comment:
I would still add a small comment here because this is kind of opaque datastructure-fu.